### PR TITLE
Fix SqlHierarchyId.GetDescendant

### PR DESCRIPTION
**Pull request by olmobrutall**
Hi dotMorten, 

I think the fix for #21 introduced some regressions, sorry that I didn't pay attention then. 

I have fixed the code and added three unit tests more. I've checked that the behavior for this four cases is identical to SQL Server native implementation. 

If you agree with the changes, would be awesome to get a new Nuget version published soon, to avoid having to distribute my own version.

Thanks a lot for your work 🙏

### CODE REVIEW

**Comment by olmobrutall** [comment]
Some test in the CI seems unstable.

Since there is no answer and I need it in my own Framework / Projects I have published a Nuget in https://www.nuget.org/packages/Unofficial.Microsoft.SqlServer.Types/ with this changes. 

The Nuget also contains the changes in https://github.com/dotMorten/Microsoft.SqlServer.Types/pull/33.

**Comment by ppasieka** [comment]
@olmobrutall I tried to use your master branch but it does not build. I believe it is because of a merge/rebase operation conflicts. Would you be able to fix it and update your PR? 

You see I'm finding your changes to solve all of my problems and it would be great to have them in this repository

**Comment by olmobrutall** [comment]
@ppasieka true, I pushed from the wrong clone :S

Now should compile.

**Comment by ppasieka** [comment]
@olmobrutall Thank you.

@dotMorten do you think we could have this PR merged and released?

File: src/Microsoft.SqlServer.Types/Microsoft.SqlServer.Types.csproj [hunk 1]
```diff
@@ -3,18 +3,18 @@
   <PropertyGroup>
     <TargetFramework>netstandard2.0</TargetFramework>
     <GeneratePackageOnBuild>true</GeneratePackageOnBuild>
-    <AssemblyVersion>1.1.0.0</AssemblyVersion>
-    <FileVersion>1.1.0.0</FileVersion>
-    <Authors>Morten Nielsen</Authors>
+    <AssemblyVersion>1.2.0.0</AssemblyVersion>
+    <FileVersion>1.2.0.0</FileVersion>
+    <Authors>Morten Nielsen (Olmo del Corral)</Authors>
```
Comment (ErikEJ) [inline src/Microsoft.SqlServer.Types/Microsoft.SqlServer.Types.csproj:8]: Why?

File: src/Microsoft.SqlServer.Types/Microsoft.SqlServer.Types.csproj [hunk 2]
```diff
@@ -3,18 +3,18 @@
...
     <Description>Allows you to use SQL Server spatial SqlHierarchy types on a machine without SQL Server installed and decode these UDT types in any .NET Standard client</Description>
     <PackageTags>SqlGeometry, SqlGeography, SqlHierarchy, SqlServer, UDT</PackageTags>
     <RepositoryUrl>https://github.com/dotMorten/Microsoft.SqlServer.Types</RepositoryUrl>
     <RepositoryType>git</RepositoryType>
     <PackageProjectUrl>https://github.com/dotMorten/Microsoft.SqlServer.Types</PackageProjectUrl>
     <PackageLicenseUrl>https://raw.githubusercontent.com/dotMorten/Microsoft.SqlServer.Types/master/LICENSE</PackageLicenseUrl>
     <Copyright>(C) 2018 - Morten Nielsen</Copyright>
-    <PackageId>dotMorten.Microsoft.SqlServer.Types</PackageId>
-    <Version>1.1.0</Version>
+    <PackageId>Unofficial.Microsoft.SqlServer.Types</PackageId>
```
Comment (ErikEJ) [inline src/Microsoft.SqlServer.Types/Microsoft.SqlServer.Types.csproj:16]: Seems like a bad idea?

File: src/Microsoft.SqlServer.Types.Tests/Microsoft.SqlServer.Types.Tests.csproj [hunk 3]
```diff
@@ -1,30 +1,30 @@
-﻿<Project Sdk="Microsoft.NET.Sdk">
-
-  <PropertyGroup>
-    <TargetFramework>netcoreapp2.0</TargetFramework>
-    <IsPackable>false</IsPackable>
-  </PropertyGroup>
-
+﻿<Project Sdk="Microsoft.NET.Sdk">
```
Comment (ErikEJ) [inline src/Microsoft.SqlServer.Types.Tests/Microsoft.SqlServer.Types.Tests.csproj:1]: Undo changes to this file

**Comment by dotMorten** [comment]
@ppsieka no this PR has too many unrelated changes that doesn't apply to issue

File: src/Microsoft.SqlServer.Types.Tests/HierarchyId/SqlHierarchyIdTests.cs [hunk 4]
```diff
@@ -1,68 +1,112 @@
...
+
+        [TestMethod]
+        public void ParseLongNodeNegative()
+        {
+            var expected = "/-281479271682120/";
+            var result = SqlHierarchyId.Parse(expected);
+            Assert.AreEqual(expected, result.ToString());
+        }
+    }
+}
```
Comment (dotMorten) [inline src/Microsoft.SqlServer.Types.Tests/HierarchyId/SqlHierarchyIdTests.cs:112]: Undo all these changes that aren't actual changes

**Comment by olmobrutall** [comment]
Sorry, I just pushed the version that I published, I was thinking this repo was dead already. 

I’ll revert the changes in the csproj files tomorrow hopefully.

**Comment by dotMorten** [comment]
>  I was thinking this repo was dead already.

Not sure what made you think that. Would be nice to ask before publishing duplicate packages on nuget.org to confuse the community.

**Comment by olmobrutall** [comment]
I made the PR and no answer for two weeks.

Then I revived my nuget because I need it for my projects and I couldn’t wait longer. One month and a a half later and no answer till today that @ppasieka revived the thread. 

Probably you were waiting for the bot to give the green light, but it worked in my machine. Maybe I pushed from the wrong repository. 

No complains to you @dotMorten, you have no obligation to spend your time maintaining a project that should be maintained by Microsoft anyway, they are getting paid for SQL Server licenses, if they don’t think that .Net Core should be a supported platform, not sure who will..

**Comment by dotMorten** [comment]
Sometimes these things notifications drop off my radar. You could try pinging me again and bumping the PR. As you know this isn't may day job and I don't get paid to do this. Especially these days where we're home all the time, and working from home, I might not be as apt to spend more time in the home office than I already did all day.

**Comment by olmobrutall** [comment]
Pushed now, tell me if there is any other issue with the code.

**Comment by olmobrutall** [comment]
Line ending change reverted

**Comment by ppasieka** [comment]
@dotMorten I see that requested changes were made to this PR. Is it possible to move this forward and merge then release?

File: src/Microsoft.SqlServer.Types.Tests/HierarchyId/SqlHierarchyIdTests.cs [hunk 5]
```diff
@@ -55,14 +56,57 @@ public void ThrowsOnNegativeAncestor()
...
+        {
+            var newSibling = SqlHierarchyId.Parse("/9/").GetDescendant(
+                SqlHierarchyId.Parse("/9/1.1/"),
+                SqlHierarchyId.Parse("/9/2/"));
+            Assert.AreEqual("/9/1.2/", newSibling.ToString());
+        }
+
         [TestMethod]
         [WorkItem(21)]
-        public void GetDescendantsOfChildren()
```
Comment (dotMorten) [inline src/Microsoft.SqlServer.Types.Tests/HierarchyId/SqlHierarchyIdTests.cs:60]: Is there a reason this test was removed? It was there specifically to test Issue #21

**Review by dotMorten — APPROVED** [review]
Re-added the missing test. Looks good

File: src/Microsoft.SqlServer.Types.Tests/HierarchyId/SqlHierarchyIdTests.cs (hunk 5 above)
Comment (olmobrutall) [reply src/Microsoft.SqlServer.Types.Tests/HierarchyId/SqlHierarchyIdTests.cs:60]: Now there are three different cases to cover, so I made three tests. This test was equivalent to GetDescendantsDecrementSecond

Comment (dotMorten) [reply src/Microsoft.SqlServer.Types.Tests/HierarchyId/SqlHierarchyIdTests.cs:60]: That's fine but no reason to remove existing testing.

**Pull request merged**